### PR TITLE
Add types declaration file

### PR DESCRIPTION
**Pull request by horaklukas**
🏷️ Add types declaration file

### CODE REVIEW

File: petrus.d.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    export function configure(config: {
+        reducerKey: string;
+        handlers: {
+            authenticate: (payload: any) => Generator<any, Pick<PetrusState, 'tokens' | 'user'>>;
+            refreshTokens: (tokens: TokensState) => Generator<any, TokensState>;
+            getAuthUser: (tokens: TokensState) => Generator<any, any>;
+        };
+        initialState: Partial<PetrusState>;
+        tokens: {
```
Comment (cermakjiri) [inline petrus.d.ts:88]:
```suggestion
        tokens?: {
```

File: petrus.d.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    export type RootSaga = () => Generator<any, void, unknown>;
+
+    export function configure(config: {
+        reducerKey: string;
+        handlers: {
+            authenticate: (payload: any) => Generator<any, Pick<PetrusState, 'tokens' | 'user'>>;
+            refreshTokens: (tokens: TokensState) => Generator<any, TokensState>;
+            getAuthUser: (tokens: TokensState) => Generator<any, any>;
+        };
+        initialState: Partial<PetrusState>;
```
Comment (cermakjiri) [inline petrus.d.ts:87]:
```suggestion
        initialState?: Partial<PetrusState>;
```

File: petrus.d.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        CombinedState<{
+            api: Reducer<ApiState>;
+            entities: PetrusState;
+        }>
+    >;
+
+    export type RootSaga = () => Generator<any, void, unknown>;
+
+    export function configure(config: {
+        reducerKey: string;
```
Comment (cermakjiri) [inline petrus.d.ts:81]:
```suggestion
        reducerKey?: string;
```

File: petrus.d.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            getAuthUser: (tokens: TokensState) => Generator<any, any>;
+        };
+        initialState: Partial<PetrusState>;
+        tokens: {
+            applyAccessTokenExternally?: boolean;
+            requestDurationEstimate?: boolean;
+            minRequiredExpiration?: boolean;
+            checkTokenExpirationOnTabFocus?: boolean;
+        };
+        logger: Pick<typeof console, 'info' | 'debug' | 'warn' | 'error'>;
```
Comment (cermakjiri) [inline petrus.d.ts:94]:
```suggestion
        logger?: Pick<typeof console, 'info' | 'debug' | 'warn' | 'error'>;
```

File: petrus.d.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            get(key: string): Promise<any>;
+            set(key: string, value: any): Promise<void>;
+            remove(key: string): Promise<void>;
+        };
+        sessionStorage: {
+            get(key: string): void;
+            set(key: string, values: any): void;
+            remove(key: string): void;
+        };
+        reset: {
```
Comment (cermakjiri) [inline petrus.d.ts:129]: what's `reset` storage driver?

**Review by cermakjiri — CHANGES_REQUESTED** [review]

File: petrus.d.ts (hunk 5 above)
Comment (horaklukas) [reply petrus.d.ts:129]: don't know, you tell me 😄 

https://github.com/AckeeCZ/petrus/blob/master/src/config/storageDrivers/reset.js

File: petrus.d.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        token: string;
+        expiration?: string;
+    }
+
+    interface TokensState {
+        accessToken?: Token;
+        refreshToken?: Token;
+    }
+
+    export function getAccessToken(): Promise<TokensState['accessToken']>;
```
[the remark this answers is not in the document]
Comment (horaklukas) [reply petrus.d.ts:58]: you're right. We use it as an async function in FS so I didn't notice it's generator. Will change

**Pull request merged**